core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingSimulator.kt

```diff
@@ -92,21 +102,37 @@ data class LimitOfAuthority(
 data class EndOfAuthority(
     val offsetEOA: Offset<TravelledPath>,
     val offsetSVL: Offset<TravelledPath>?,
-    val usedCurveType: BrakingType
+    val usedCurveType: BrakingType,
+    val eoaType: EoaType = EoaType.STOP,
 ) : Comparable<EndOfAuthority> {
     init {
         if (offsetSVL != null) assert(offsetSVL >= offsetEOA)
     }
 
     override fun compareTo(other: EndOfAuthority): Int {
         if (offsetEOA != other.offsetEOA) return offsetEOA.compareTo(other.offsetEOA)
-        if (offsetSVL == null && other.offsetSVL == null) return 0
-        if (offsetSVL == null) return -1
-        if (other.offsetSVL == null) return 1
-        return offsetSVL.compareTo(other.offsetSVL)
+        return if (offsetSVL == other.offsetSVL) {
+            if (usedCurveType == other.usedCurveType && eoaType == other.eoaType) {
+                0
+            } else {
+                1
+            }
+        } else if (offsetSVL == null) {
+            -1
+        } else if (other.offsetSVL == null) {
+            1
+        } else {
+            offsetSVL.compareTo(other.offsetSVL)
+        }
```

Comment:
The comparison is not commutative on `usedCurveType` and `eoaType` (depending if an EndOfAuthority is `this` or `other`):
```suggestion
        if (offsetSVL != other.offsetSVL) {
            if (offsetSVL == null) return -1
            if (other.offsetSVL == null) return 1
            return offsetSVL.compareTo(other.offsetSVL)
        }
        if (usedCurveType != other.usedCurveType) {
            return usedCurveType.compareTo(other.usedCurveType)
        }
        if (eoaType != other.eoaType) {
            return eoaType.compareTo(other.eoaType)
        }
        return 0
```

Reply:
See https://github.com/OpenRailAssociation/osrd/pull/12457/commits/ae9ef40ae4996c3e38519a9cd4f5fd683d513732.